src/key_packages/mod.rs

```diff
@@ -267,29 +267,65 @@ impl KeyPackageBundle {
     ///
     /// Note that the capabilities extension gets added automatically, based on
     /// the configuration. The ciphersuite for this key package bundle is the
-    /// first one in the `ciphersuites` list. Only the ciphersuites listed in
-    /// `ciphersuites` are enabled in the capabilities extension of the key package.
+    /// first one in the `ciphersuites` list. If a capabilities extension is
+    /// included in the extensions, its supported ciphersuites have to match the
+    /// `ciphersuites` list.
     ///
     /// Returns a new `KeyPackageBundle`.
     pub fn new_with_keypair(
         ciphersuites: &[CiphersuiteName],
         credential_bundle: &CredentialBundle,
-        extensions: Vec<Box<dyn Extension>>,
+        mut extensions: Vec<Box<dyn Extension>>,
         key_pair: HPKEKeyPair,
         leaf_secret: Secret,
     ) -> Result<Self, ConfigError> {
         debug_assert!(!ciphersuites.is_empty());
-        let capabilities_extension = CapabilitiesExtension::new(None, Some(ciphersuites), None);
-        let mut final_extensions: Vec<Box<dyn Extension>> = vec![Box::new(capabilities_extension)];
 
+        // Remove any duplicate extensions
```

Comment:
Hm, so this is an API decision we have to take here. Right now we simply remove duplicates and continue with the lucky extension that we didn't delete. I think this is a little too random for the caller.
I think we should return an error if there is a duplicate in there. So something like 
```
let extensions_len = extensions.len();
extensions.sort();
extensions.dedup();
if extensions.len() != extensions_len {
  return Err(ConfigError::InvalidExtensions);
}
```